### ISSUE TITLE
update: notifications dissapear after an app update

### ISSUE THREAD

**Issue by tiagogaspar8** [signalapp/Signal-Android#11963]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

When I receive a message a notification is shown, as expected, but when the app is manually updated manually or automatically in the background by Google play store the notifications dissapear and are never shown again.

### Steps to reproduce

- Recieve a message (and a notification is shown
- Go to the Google play store and update signal (signal is updated)
- The previous message notification dissapears and is never shown again

**Actual result:** 

The notification dissapears and is never shown again

**Expected result:** 

The notification dissapears while the app is being updated (cause it needs to be closed) and once it is updated it searches for unread notifications and sends them again to the notification bar

### Screenshots

No screenshots are applicable

### Device info

**Device:** OnePlus 8
**Android version:** 11
**Signal version:** 5.31.4

### Link to debug log
https://debuglogs.org/3a31f400c73f581fa95282ca177c3ceebbda096589f50bbd9abf66aaacb8e720

**Comment by tiagogaspar8** [comment]
Hi all,

This issue should be kept open, the issue is still present in the latest update.

Thanks!

**Comment by AmirGamilDev** [comment]
Can confirm still occurring for me also. Also worth noting that other messaging apps handle this as the expected behaviour.

**Comment by mvforell** [comment]
I can confirm this still occurs with version 6.40.5.

**Comment by AmirGamilDev** [comment]
I agree. I actually have auto update off on Signal just in case. And some
phones have an auto restart feature which would result in a similar result.

We appreciate it may not be an easy fix but it is important. Thanks for all
your hard work making Signal as awesome as it is! 👍👏

--------

**Comment by Aevrin** [comment]
Can we add the notifications label to this issue?
